modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/util/KBDropdownItemsProvider.java

```diff
@@ -426,7 +506,22 @@ public List<DropdownItem> getKBFolderDropdownItems(KBFolder kbFolder) {
 					).setActionName(
 						"/knowledge_base/delete_kb_folder"
 					).setRedirect(
-						_currentURL
+						() -> {
+							if (ListUtil.isNotEmpty(selectedItemAncestorIds)) {
+								if (selectedItemAncestorIds.contains(
+										kbFolder.getKbFolderId())) {
+
+									return _createKbFolderRenderURL(
+										kbFolder.getParentKBFolderId());
+								}
+							}
+							else if (_isKBFolderSelected(kbFolder)) {
+								return _createKbFolderRenderURL(
+									kbFolder.getParentKBFolderId());
+							}
+
+							return _currentURL;
```

Comment:
See my comment on the equivalent code for kb articles.

Reply:
The conditions are not disjoint, so simplification cannot be applied.